File: clean_routes.py
import re

with open('frontend/src/App.jsx', 'r') as f:
    content = f.read()

# Routes to remove
routes = [
    r'master/managers',
    r'master/bank-accounts',
    r'master/bank-accounts/:id',
    r'settings/branding',
    r'settings/ui-settings',
    r'settings/smtp-email',
    r'settings/notifications',
    r'transactions/transfer-voucher/payment',
    r'transactions/transfer-voucher/payment/:id',
    r'master/demands',
    r'master/demands/:id',
    r'master/no-interest-members',
    r'master/no-interest-members/:id'
]

imports = [
    'ManagerMasterPage',
    'BankAccountsPage',
    'BankAccountDetailPage',
    'BrandingPage',
    'UiSettingsPage',
    'SmtpEmailPage',
    'NotificationSettingsPage',
    'TransferVoucherPaymentWorkspacePage',
    'TransferVoucherPaymentWorkspaceDetailPage',
    'DemandsPage',
    'DemandDetailPage',
    'NoInterestMembersPage',
    'NoInterestMemberDetailPage'
]

# We want to match `<Route ... />` or `<Route ...> ... </Route>`
# Since JSX is nested, a simple regex might fail on arbitrary nesting, but for <Route> it's well defined.
# A Route element starts with `<Route` and ends with either `/>` or `</Route>`.
# We'll use a regex that handles both correctly without crossing `<Route` boundaries if possible.
# Actually, since these routes don't contain other `<Route>` inside them, we can safely match up to the first `/>` or `</Route>` that comes AFTER the path="...".
# We must find `<Route` then any chars except `<Route` (wait, no, `<Route` can contain other tags, but not `<Route`).
# So: <Route(?:(?!<Route).)*?path="THE_PATH"(?:(?!<Route).)*?(?:/>|</Route>)
# This uses negative lookahead to ensure we don't cross another <Route

for route in routes:
    # Handle single or double quotes
    pattern = r'<Route(?:(?!<Route)[\s\S])*?path=["\']' + route + r'["\'](?:(?!<Route)[\s\S])*?(?:/>|</Route>)'
    content = re.sub(pattern, '', content)

# Also remove the imports
for imp in imports:
    pattern = r'^import\s+.*?' + imp + r'.*?;\n?'
    content = re.sub(pattern, '', content, flags=re.MULTILINE)

# Remove extra empty lines
content = re.sub(r'\n\s*\n\s*\n', '\n\n', content)

with open('frontend/src/App.jsx', 'w') as f:
    f.write(content)
print("Cleaned App.jsx with Python!")

File: clean_routes.js
const fs = require('fs');

const filesToRemove = [
  'frontend/src/pages/master/ManagerMasterPage.jsx',
  'frontend/src/pages/master/bank-accounts',
  'frontend/src/pages/settings/BrandingPage.jsx',
  'frontend/src/pages/settings/UiSettingsPage.jsx',
  'frontend/src/pages/settings/SmtpEmailPage.jsx',
  'frontend/src/pages/settings/notifications',
  'frontend/src/pages/transactions/transfer-voucher/paymentWorkspace.jsx',
  'frontend/src/pages/transactions/transfer-voucher/paymentWorkspaceDetail.jsx',
  'frontend/src/pages/transactions/interest/receiptWorkspace.jsx',
  'frontend/src/pages/transactions/interest/receiptWorkspaceDetail.jsx',
  'frontend/src/pages/transactions/interest/interestRecvMember.jsx',
  'frontend/src/pages/transactions/interest/interestRecvEmployee.jsx',
  'frontend/src/pages/reports/SummaryMonthlyPage.jsx',
  'frontend/src/pages/master/demands',
  'frontend/src/pages/master/no-interest-members',
];

const routesToRemove = [
  'master/managers',
  'master/bank-accounts',
  'settings/branding',
  'settings/ui-settings',
  'settings/smtp-email',
  'settings/notifications',
  'transactions/transfer-voucher/transfer-voucher-payment',
  'transactions/transfer-voucher/transfer-voucher-receipt',
  'transactions/interest/interest-recv-member',
  'transactions/interest/interest-recv-employee',
  'reports/summary-monthly',
  'master/demands',
  'master/no-interest-members'
];

let appJsx = fs.readFileSync('frontend/src/App.jsx', 'utf8');

// Also try to find exact imports to remove
const importsToRemove = [
  'ManagerMasterPage',
  'BankAccountsPage',
  'BankAccountDetailPage',
  'BrandingPage',
  'UiSettingsPage',
  'SmtpEmailPage',
  'NotificationSettingsPage',
  'TransferVoucherPaymentWorkspacePage',
  'TransferVoucherPaymentWorkspaceDetailPage',
  'ReceiptVoucherWorkspacePage',
  'ReceiptVoucherWorkspaceDetailPage',
  // Not sure what the exact component names were for the interest ones, but we will remove the routes at least.
];

// Remove routes
routesToRemove.forEach(route => {
  const escapedRoute = route.split('/').join('\\/');
  // Match single line <Route path="..." ... />
  const regex1 = new RegExp(`<Route[^>]*path=["']\\/?${escapedRoute}(\\/\\:id)?["'][^>]*\\/>`, 'g');
  appJsx = appJsx.replace(regex1, '');
  // Match multi-line <Route path="..."> ... </Route>
  const regex2 = new RegExp(`<Route[^>]*path=["']\\/?${escapedRoute}(\\/\\:id)?["'][\\s\\S]*?<\\/Route>`, 'g');
  appJsx = appJsx.replace(regex2, '');
});

// Remove imports
importsToRemove.forEach(imp => {
  const impRegex = new RegExp(`^import.*${imp}.*$;?`, 'gm');
  appJsx = appJsx.replace(impRegex, '');
});

// Cleanup empty lines
appJsx = appJsx.replace(/\n\s*\n\s*\n/g, '\n\n');

fs.writeFileSync('frontend/src/App.jsx', appJsx, 'utf8');
console.log("App.jsx cleaned");

// Delete files
filesToRemove.forEach(f => {
  if (fs.existsSync(f)) {
    fs.rmSync(f, { recursive: true, force: true });
    console.log(`Deleted ${f}`);
  }
});
